enterprise/com/src/main/java/org/neo4j/com/storecopy/ToFileStoreWriter.java

```diff
@@ -23,46 +23,77 @@
 import java.io.IOException;
 import java.io.RandomAccessFile;
 import java.nio.ByteBuffer;
-import java.nio.channels.FileChannel;
 import java.nio.channels.ReadableByteChannel;
+import java.nio.channels.WritableByteChannel;
+import java.util.List;
+import java.util.Optional;
+
+import org.neo4j.io.pagecache.PageCache;
+import org.neo4j.io.pagecache.PagedFile;
+import org.neo4j.kernel.impl.store.StoreType;
+
+import static java.nio.file.StandardOpenOption.CREATE;
+import static java.nio.file.StandardOpenOption.WRITE;
+import static org.neo4j.kernel.impl.store.StoreType.isStoreType;
+import static org.neo4j.kernel.impl.store.format.RecordFormat.NO_RECORD_SIZE;
 
 public class ToFileStoreWriter implements StoreWriter
 {
     private final File basePath;
     private final StoreCopyClient.Monitor monitor;
+    private final PageCache pageCache;
+    private final List<FileMoveAction> fileMoveActions; // Todo need thread safe?
```

Comment:
No, I don't think this needs to be thread safe. The files are streaming over one at a time, controlled on our side by a single thread.